HeterogeneousCore/CUDAUtilities/interface/CUDAStreamCache.h

```diff
@@ -1,30 +1,43 @@
 #ifndef HeterogeneousCore_CUDAUtilities_CUDAStreamCache_h
 #define HeterogeneousCore_CUDAUtilities_CUDAStreamCache_h
 
-#include <memory>
+#include <vector>
 
-#include <cuda/api_wrappers.h>
+#include <cuda_runtime.h>
 
 #include "FWCore/Utilities/interface/ReusableObjectHolder.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/SharedStreamPtr.h"
 
 class CUDAService;
 
 namespace cudautils {
   class CUDAStreamCache {
   public:
+    using BareStream = SharedStreamPtr::element_type;
```

Comment:
is this a plain `cudaStream_t` ?

Reply:
It is the pointed-to type of `cudaStream_t` (`CUstream_st` or something), but I'd rather not explicitly use types from the driver API.

Reply:
ditto